addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/handler/RouteHandler.java

```diff
@@ -0,0 +1,278 @@
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onebusaway.config.ChannelConfig;
+import org.openhab.binding.onebusaway.config.RouteConfiguration;
+import org.openhab.binding.onebusaway.handler.ObaStopArrivalResponse.ArrivalAndDeparture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Function;
+
+/**
+ * The {@link RouteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Shawn Wilsher - Initial contribution
+ */
+public class RouteHandler extends BaseThingHandler implements RouteDataListener {
+
+    public static final ThingTypeUID SUPPORTED_THING_TYPE = THING_TYPE_ROUTE;
+
+    private RouteConfiguration config;
+    private Logger logger = LoggerFactory.getLogger(RouteHandler.class);
+    private List<ScheduledFuture<?>> scheduledFutures = new CopyOnWriteArrayList<>();
+
+    public RouteHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH == command) {
+            logger.debug("Refreshing {}...", channelUID);
+            switch (channelUID.getId()) {
+                case CHANNEL_ID_ARRIVAL:
+                case CHANNEL_ID_DEPARTURE:
+                case CHANNEL_ID_UPDATE:
+                    StopHandler stopHandler = getStopHandler();
+                    if (stopHandler != null) {
+                        stopHandler.forceUpdate();
+                    }
+                    break;
+                default:
+                    logger.warn("Unhandled command for channel: {} with comamnd: {}", channelUID.getId(),
+                            command.toString());
+            }
+        } else {
+            logger.warn("The OneBusAway binding is a read-only binding and can not handle commands.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.warn("The OneBusAway binding is a read-only binding and can not handle channel updates.");
+        super.handleUpdate(channelUID, newState);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.info("Initializing OneBusAway route stop...");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Checking configuration...");
+
+        config = loadAndCheckConfiguration();
+        if (config == null) {
+            logger.debug("Initialization of OneBusAway route stop failed!");
+            return;
+        }
+        StopHandler stopHandler = getStopHandler();
+        if (stopHandler != null) {
+            stopHandler.registerRouteDataListener(this);
```

Comment:
I'm not quite sure I understand your comment.  Do you mean that it's never marked as online?  If that's what you meant, I can explain (and I'll add a comment in the code since it wasn't obvious during code review here).

The `stopHandler` here will notify this thing when it has data for the stop (it is the bridge for this thing).  The bridge is online once it loads data (see `StopHandler`), and then when we get initialized, we call `registerRouteDataListener`, which will notify this class (via `onNewRouteData`), which then marks things as online.